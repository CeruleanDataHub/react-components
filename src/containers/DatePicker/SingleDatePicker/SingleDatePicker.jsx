import "react-dates/initialize";
import "react-dates/lib/css/_datepicker.css";

import moment from "moment";
import PropTypes from "prop-types";
import React, { useState } from "react";
import { SingleDatePicker as DatePicker } from "react-dates";
import momentPropTypes from "react-moment-proptypes";
import styled from "styled-components";

import { Icon } from "../../Icon/Icon";
import { Typography } from "../../Typography/Typography";
import { SingleDatePickerStyles } from "../datePickerStyles";

const Container = styled.div`
  ${SingleDatePickerStyles}
`;

const CalendarIcon = styled.span`
  font-size: 1.25em;
`;

export const SingleDatePicker = ({
  name,
  startDate,
  dateFormat,
  monthsShown
}) => {
  const [dateValue, setDateValue] = useState(startDate);
  const [isFocused, setIsFocused] = useState(false);

  return (
    <Container>
      <Typography fontFamily="openSans">
        <DatePicker
          date={dateValue}
          onDateChange={date => setDateValue(date)}
          focused={isFocused}
          onFocusChange={({ focused }) => setIsFocused(focused)}
          numberOfMonths={monthsShown}
          firstDayOfWeek={1}
          displayFormat={dateFormat}
          customInputIcon={<Icon name="calendar" as={CalendarIcon} />}
          id={`${name}-single-datepicker`}
          hideKeyboardShortcutsPanel
        />
      </Typography>
    </Container>
  );
};

SingleDatePicker.propTypes = {
  /** Unique id for datepicker */
  name: PropTypes.string.isRequired,
  /** Starting date to be shown on date picker. Defaults to current day. */
  startDate: momentPropTypes.momentObj,
  /** How the date should be displayed. */
  dateFormat: PropTypes.string,
  /** How many months are shown. */
  monthsShown: PropTypes.number
};

SingleDatePicker.defaultProps = {
  startDate: moment(),
  dateFormat: "DD.MM.yyyy",
  monthsShown: 1
};
