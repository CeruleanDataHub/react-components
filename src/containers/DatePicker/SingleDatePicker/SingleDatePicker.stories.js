import moment from "moment";
import React from "react";

import { SingleDatePicker } from "./SingleDatePicker";

export default {
  title: "SingleDatePicker",
  component: SingleDatePicker
};

export const SingleDatePickerStory = () => (
  <div style={{ minHeight: 400 }}>
    <SingleDatePicker name="default" />
  </div>
);

export const SingleDatePickerWithCustomDateFormatStory = () => (
  <div style={{ minHeight: 400 }}>
    <SingleDatePicker name="dateformat" dateFormat="MM/DD/yyyy" />
  </div>
);

export const SingleDatePickerTwoMonthsShownStory = () => (
  <div style={{ minHeight: 400 }}>
    <SingleDatePicker name="months" monthsShown={2} />
  </div>
);

export const SingleDatePickerWithCustomStartDateStory = () => (
  <div style={{ minHeight: 400 }}>
    <SingleDatePicker name="startDate" startDate={moment("2016-06-01")} />
  </div>
);
