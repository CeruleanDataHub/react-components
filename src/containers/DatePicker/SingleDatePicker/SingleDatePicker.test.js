import moment from "moment";
import React from "react";
import renderer from "react-test-renderer";

import { SingleDatePicker } from "./SingleDatePicker";

const mockDate = new Date("2020-06-19");
const RealDate = Date;

describe("SingleDatePicker", () => {
  beforeAll(() => {
    global.Date = class extends RealDate {
      constructor() {
        super();
        return mockDate;
      }
    };
  });

  afterAll(() => {
    global.Date = RealDate;
  });

  it("should render", () => {
    const component = renderer.create(<SingleDatePicker />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with custom date format", () => {
    const component = renderer.create(
      <SingleDatePicker dateFormat="MM/DD/yyyy" />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render with two months shown", () => {
    const component = renderer.create(<SingleDatePicker monthsShown={2} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe("SingleDatePicker with given date", () => {
  it("should render with given startDate", () => {
    const component = renderer.create(
      <SingleDatePicker startDate={moment("2020-01-01")} />
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
