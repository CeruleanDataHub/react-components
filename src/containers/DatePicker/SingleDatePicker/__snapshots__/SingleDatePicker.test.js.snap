// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleDatePicker dark should render 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: none;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #0f181c;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #0f181c;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #0f181c;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #0f181c;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #0f181c;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #0f181c;
  color: #999;
  border: 1px solid #000000;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #90ee7e;
  color: #000000;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-single-datepicker"
              name="test-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker dark should render with custom date format 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: none;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #0f181c;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #0f181c;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #0f181c;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #0f181c;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #0f181c;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #0f181c;
  color: #999;
  border: 1px solid #000000;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #90ee7e;
  color: #000000;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-single-datepicker"
              name="test-format-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker dark should render with two months shown 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: none;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #0f181c;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #0f181c;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #0f181c;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #0f181c;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #0f181c;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #0f181c;
  color: #999;
  border: 1px solid #000000;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #90ee7e;
  color: #000000;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-months-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-months-single-datepicker"
              name="test-months-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-months-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker light should render 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: 0 0 7px #e0e1e4;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #fff;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #fff;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #fff;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #fff;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #fff;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #fff;
  color: #999;
  border: 1px solid #dedede;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #2a908e;
  color: #ffffff;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-single-datepicker"
              name="test-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker light should render with custom date format 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: 0 0 7px #e0e1e4;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #fff;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #fff;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #fff;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #fff;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #fff;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #fff;
  color: #999;
  border: 1px solid #dedede;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #2a908e;
  color: #ffffff;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-format-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-format-single-datepicker"
              name="test-format-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="06/19/2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-format-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker light should render with two months shown 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: 0 0 7px #e0e1e4;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #fff;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #fff;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #fff;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #fff;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #fff;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #fff;
  color: #999;
  border: 1px solid #dedede;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #2a908e;
  color: #ffffff;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-months-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-months-single-datepicker"
              name="test-months-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-months-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker with given date should render with given startDate 1`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: 0 0 7px #e0e1e4;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #fff;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #fff;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #fff;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #fff;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #fff;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #fff;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #fff;
  color: #999;
  border: 1px solid #dedede;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #2a908e;
  color: #ffffff;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-date-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-date-single-datepicker"
              name="test-date-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-date-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`SingleDatePicker with given date should render with given startDate 2`] = `
.c3 {
  text-align: center;
}

.c1 {
  font-family: "Open Sans",sans-serif;
  font-size: 16px;
  line-height: 20px;
  color: unset;
}

.c0 .SingleDatePicker {
  color: #999;
  cursor: pointer;
  box-shadow: none;
}

.c0 .SingleDatePicker .SingleDatePickerInput {
  background: #0f181c;
  border-radius: 4px;
  border: none;
}

.c0 .SingleDatePicker .DateInput {
  color: #999;
}

.c0 .SingleDatePicker .DateInput_fang {
  display: none;
}

.c0 .SingleDatePicker .DateInput_fangStroke {
  fill: #0f181c;
}

.c0 .SingleDatePicker .DateInput_input {
  color: inherit;
  background-color: #0f181c;
  border-radius: 0 4px 4px 0;
}

.c0 .SingleDatePicker .DateInput_input__focused {
  border-bottom-color: transparent;
}

.c0 .SingleDatePicker .DayPicker {
  box-shadow: none;
  background: #0f181c;
  border-radius: 0;
}

.c0 .SingleDatePicker .DayPicker_transitionContainer {
  border-radius: 0;
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonthGrid {
  background: #0f181c;
}

.c0 .SingleDatePicker .CalendarMonth {
  background: #0f181c;
  padding: 0 13px 10px 13px;
}

.c0 .SingleDatePicker .CalendarMonth_caption {
  color: #999;
  padding-bottom: 42px;
}

.c0 .SingleDatePicker .DayPickerNavigation_button__default {
  background: transparent;
  border: none;
}

.c0 .SingleDatePicker .CalendarDay__default {
  background: #0f181c;
  color: #999;
  border: 1px solid #000000;
}

.c0 .SingleDatePicker .CalendarDay__default:hover {
  background: #ffffff;
  color: #000000;
}

.c0 .SingleDatePicker .CalendarDay__selected,
.c0 .SingleDatePicker .CalendarDay__selected_span,
.c0 .SingleDatePicker .CalendarDay__hovered_span:hover,
.c0 .SingleDatePicker .CalendarDay__hovered_span {
  background: #90ee7e;
  color: #000000;
}

.c2 {
  font-size: 1.25em;
}

<div
  className="c0"
>
  <div
    className="c1"
    fontFamily="openSans"
    size="normal"
  >
    <div
      className="SingleDatePicker SingleDatePicker_1"
    >
      <div>
        <div
          className="SingleDatePickerInput SingleDatePickerInput_1 SingleDatePickerInput__withBorder SingleDatePickerInput__withBorder_2"
        >
          <button
            className="SingleDatePickerInput_calendarIcon SingleDatePickerInput_calendarIcon_1"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <span
              className="c2 c3 lni lni-calendar"
            />
          </button>
          <div
            className="DateInput DateInput_1"
          >
            <input
              aria-describedby="DateInput__screen-reader-message-test-date-single-datepicker"
              aria-label="Date"
              autoComplete="off"
              className="DateInput_input DateInput_input_1"
              disabled={false}
              id="test-date-single-datepicker"
              name="test-date-single-datepicker"
              onChange={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder="Date"
              readOnly={false}
              required={false}
              type="text"
              value="19.06.2020"
            />
            <p
              className="DateInput_screenReaderMessage DateInput_screenReaderMessage_1"
              id="DateInput__screen-reader-message-test-date-single-datepicker"
            >
              Navigate forward to interact with the calendar and select a date. Press the question mark key to get the keyboard shortcuts for changing dates.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
